modules/gdscript/gdscript.cpp

```diff
@@ -2458,30 +2459,99 @@ String GDScriptLanguage::get_global_class_name(const String &p_path, String *r_b
 
 	GDScriptParser parser;
 	err = parser.parse(source, p_path, false);
-	if (err) {
-		return String();
-	}
-
-	GDScriptAnalyzer analyzer(&parser);
-	err = analyzer.resolve_inheritance();
-	if (err) {
-		return String();
-	}
 
 	const GDScriptParser::ClassNode *c = parser.get_tree();
-
-	if (r_base_type) {
-		*r_base_type = c->get_datatype().native_type;
-	}
+	if (!c) {
+		return String(); // No class parsed.
+	}
+
+	/* **WARNING**
+	 *
+	 * This function is written with the goal to be *extremely* error tolerant, as such
+	 * it should meet the following requirements:
+	 *
+	 * - It must not rely on the analyzer, because at the time global classes are parsed,
+	 *   the dependencies may not be present yet, hence the function will fail (which is unintended).
+	 * - It must not fail even if the parsing fails, because even if the file is broken,
+	 *   it should attempt its best to retrieve the inheritance metadata.
+	 *
+	 * Before changing this function, please ask the current maintainer of EditorFileSystem.
+	 */
 
 	if (r_icon_path) {
 		if (c->icon_path.is_empty() || c->icon_path.is_absolute_path()) {
-			*r_icon_path = c->icon_path.simplify_path();
+			*r_icon_path = c->icon_path;
```

Comment:
This is a minor remark, but I think `simplify_path()` at least does no harm here. :slightly_smiling_face: